### PR TITLE
Install Script

### LINKED ISSUE #705: One-Liner install script
## Summary

It would be nice to give users an instant way to get a node up and running directly on Readme.md without cluttering it up with a code snippet. 

## Problem Definition

Currently, setting up Gaia takes too many discrete steps and involves visiting numerous repositories for information.  

## Initial Proposal (scrapping)

Adapt the [Galaxy installer](https://github.com/galaxypi/galaxy/blob/master/install.sh) for use with Gaia.

- [ ] ~~Script should work on mac and windows~~

## 2nd Proposal

After looking this over, to be cross-platform, it seems to make the most sense to make a script that serves as a container's entrypoint, or start packaging Gaia for choco, brew, and apt, at a minimum.   

contrib/instagaia/instagaia.sh

```bash
#!/bin/sh
#give the node a random moniker
gaiad init instagaia$RANDOM
# Start the node with the default set of seeds
gaiad start --p2p.seeds bf8328b66dceb4987e5cd94430af66045e59899f@public-seed.cosmos.vitwit.com:26656,cfd785a4224c7940e9a10f6c1ab24c343e923bec@164.68.107.188:26656,d72b3011ed46d783e369fdf8ae2055b99a1e5074@173.249.50.25:26656,ba3bacc714817218562f743178228f23678b2873@public-seed-node.cosmoshub.certus.one:26656,3c7cad4154967a294b3ba1cc752e40e8779640ad@84.201.128.115:26656
```

contrib/instagaia/Dockerfile
```Dockerfile
FROM tendermint/gaia

USER gaia

COPY contrib/instagaia/instagaia.sh /gaia/instagaia.sh

RUN wget https://github.com/cosmos/mainnet/raw/master/genesis.cosmoshub-4.json.gz && \
           gzip -d genesis.cosmoshub-4.json.gz && \
           mv genesis.cosmoshub-4.json ~/gaia/genesis.json


EXPOSE 46657
EXPOSE 46656
EXPOSE 1317

ENTRYPOINT ["/bin/sh", "/gaia/instagaia.sh"]
```

This will give us something that can go on Readme.md which will look like this:

```bash
docker run tendermint/instagaia
```

- [x] Runs on Windows, Mac and Linux

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

### PR DESCRIPTION
**Pull request by faddat**
Closes: #705 

## Description

Adds an install.sh to the repository root and a snippet for using it to readme.md
______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/main/README.md#merging-a-pr))

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+# Banner 'cause why not right?'
+echo -e "
+  _____          _____          
+ / ____|   /\   |_   _|   /\    
+| |  __   /  \    | |    /  \   
+| | |_ | / /\ \   | |   / /\ \  
+| |__| |/ ____ \ _| |_ / ____ \ 
+ \_____/_/    \_\_____/_/    \_\
+                                
+"
```
Comment (tac0turtle) [inline install.sh:13]: love this

**Comment by faddat** [comment]
~/gaia/gaiad

or

/usr/local/bin/gaiad per #495 ?

I'm thinking that "on the path" is better-- this way, users don't even need to worry about Go.

**Comment by faddat** [comment]
Making this universal is proving quite difficult.  I've outlined a second approach that uses docker in #705

**Comment by faddat** [comment]
I'm going to close this but could reopen it if we feel like we really need a bash script.

Now that Docker is building multi-platform it seems to make more sense to allow Docker to do that. I did find a better approach than the script that I wrote in 2018 along with @vsund and @guylepage3, which is called go downloader (part of goreleaser) and they're using it at Akash. 

Thing is it is no longer maintained upstream, and it still would not give us windows support.

So I've got a couple of points on this, first of all it seems like we could reasonably add support for snap packages, that'd give us a snap install gaia or snap install gaiad.

We could probably do the same thing with Homebrew.

Furthermore we could probably do the same thing with Choco for windows.  

But for now, I think it makes the most sense to ship as a container with a volume mount, that way upgrades are smooth, No matter what context gaia is being deployed in.

**Pull request closed without merging**